### PR TITLE
Handle partial search result with point in time

### PR DESCRIPTION
**Pull request by dnhatn**
Today, a search request with PIT would fail immediately if any associated indices or nodes are gone, which is inconsistent when `allow_partial_search_results` is true.

Relates #81256

### CODE REVIEW

**Review by ywelsch — COMMENTED** [review]
This PR changes two things:

- Make PIT work with allow_partial_results = true (previously it would fail)
- Change http response code when allow_partial_results = false

While the former looks ok for me, I'm worried a bit about the BWC implications of the latter, especially as this is being backported to 7.16.

Should we only do 1) here for the backport and follow-up with 2) in a separate PR?

**Comment by dnhatn** [comment]
> Change http response code when allow_partial_results = false

I pushed https://github.com/elastic/elasticsearch/pull/81349/commits/8b1be39a653a70a07d9d3e182b67ba0537f1edb6 to revert the current behaviour when `allow_partial_results` is false. I am not sure the consequence of this as allow_partial_results defaults to true.

File: docs/reference/search/point-in-time-api.asciidoc [hunk 1]
```diff
@@ -54,7 +54,7 @@ POST /_search <1>
     }
 }
 --------------------------------------------------
-// TEST[catch:missing]
+// TEST[catch:unavailable]
```
Comment (dnhatn) [inline docs/reference/search/point-in-time-api.asciidoc:57]: We have to change this as the `allow_partial_results` defaults to true.

**Comment by dnhatn** [comment]
@elasticmachine update branch

**Review by ywelsch — APPROVED** [review]
LGTM

**Comment by dnhatn** [comment]
run elasticsearch-ci/part-2

**Comment by dnhatn** [comment]
Thanks Yannick!

**Pull request merged**